decisions/0001-file-sharing.md

```diff
@@ -0,0 +1,103 @@
+
+The raw read data of a sample is significantly larger than a consensus sequence. For example, a consensus sequence of a SARS-CoV-2 sample is about 30 KB, while the raw data is around hundred MB to a few GB. We therefore need an approach to store and share large files.
+
+From the user perspective, depending on the instance, it should be possible to upload metadata, consensus sequences and/or raw data. The consensus sequences are, as now, stored in a FASTA. The raw data can be a FASTQ, a BAM file or something else – what exactly one has/can upload depends also on the instance. The maintainer-configurable pipeline will take the user data and return processed data: this could be only consensus sequences and metadata (i.e., some tabular data), or include some other unaligned raw data (e.g., the raw reads after dehumanizing), aligned raw reads, or other type of files. The user will be able to download all processed data. Aligned reads may be passed into SILO for fast querying.
+
+
+## Decision Drivers
+
+* Scalability of storage size: The system should be able to scale to TBs or even PBs of data files.
+* Scalability of traffic: The system should be able to handle the upload and download of large files.
+* Same data privacy, integrity, and persistence model as before: This means that the data uploaded by a group should remain private until it has been approved. Once approved, it should be guaranteed that the originally uploaded data remain available and versioned (unless explicit interventions by a maintainer).
+
+
+## Considered Options
+
+### S3 and pre-signed URLs
+
+- To upload files, the authenticated client first calls a new endpoint `/files/request-uploads`. The backend returns S3 pre-signed URLs to which the client uploads the files.
+- When the client calls `/submit` (or `/revise`), they can provide in the metadata file in the column `files` the file IDs of the associated files.
+- The backend will provide to the preprocessing pipeline pre-signed URLs to read the original files. If the pipeline would like to upload processed files, it calls the same `/files/request-uploads` endpoint.
+- During the review, the user receives pre-signed URLs to download the original and processed files.
+
+### S3 and (semi-)public write-only link
+
+- Authenticated users can see on the website information to upload to files to a shared S3 bucket.
+- They should choose a random object names to avoid conflicts with uploads of other users.
+- They state the object name in the metadata file when they submit (or revise).
+
+### User-controlled S3
+
+- Users upload files to their own S3 bucket and give Loculus read access.
```

Comment:
If WISE is the only user (and they already have an S3 this might actually be enough for the moment)

Reply:
At WISE, there is no maintained S3 instance at the moment (data are just stored on disk in the network, if I'm not mistaken). The goal of this implementation is to enable using Loculus to manage the data.